### ISSUE TITLE
 RandCropByPosNegLabeld

### ISSUE THREAD

**Issue by guigautier** [Project-MONAI/MONAI#1063]
Hi, I try to use  RandCropByPosNegLabeld to generate my training dataset. I can't generate the same crop between the image and the label.
I load a nifti file and save it with the writer after transform

This is my code to reproduce : 
```
            train_transforms = Compose([
                LoadNiftid(keys=['image', 'label']),
                AddChanneld(keys=['image', 'label']),
                Spacingd(keys=['image', 'label'], pixdim=(1, 1, 1)),

                Orientationd(keys=['image', 'label'], axcodes='RAS'),
                # Resized(keys=['image', 'label'], spatial_size=(256, 256, 256)),
                NormalizeIntensityd(keys=['image']),
                RandCropByPosNegLabeld(keys=['image', 'label'], label_key="label",
                                       spatial_size=(128, 128,
                                                     128), num_samples=4),
                ToTensord(keys=['image', 'label'])
            ])

            train_transforms.set_random_state(seed=0)
            torch.manual_seed(0)
 
self.train_ds = monai.data.Dataset(data=train_files, transform=train_transforms)

image = self.train_ds[0][0]["image"][:, :, :, :].detach().cpu()
nifti_saver = monai.data.NiftiSaver(output_postfix="i")
nifti_saver.save(image)
label = self.train_ds[0][0]["label"][:, :, :, :].detach().cpu()
nifti_saver = monai.data.NiftiSaver(output_postfix="l")
nifti_saver.save(label)
```

Link to the image and label overlay : https://ibb.co/s1bxxjG

Thanks

**Comment by wyli** [comment]
this is a good question, this get item `self.train_ds[0]` will actually run the randomised transforms, so the results will be different.

so you could do:
```py
data = self.train_ds[0][0]
image = data["image"][:, :, :, :].detach().cpu()
nifti_saver = monai.data.NiftiSaver(output_postfix="i")
nifti_saver.save(image)
label = data["label"][:, :, :, :].detach().cpu()
nifti_saver = monai.data.NiftiSaver(output_postfix="l")
nifti_saver.save(label)
```

some small demo for your reference:
```py
In [1]: import monai                                                                                                                                       

In [2]: def print_transform(x): 
   ...:     print("called") 
   ...:     return x 
   ...:                                                                                                                                                    

In [3]: a = monai.data.Dataset([1, 2, 3], transform=print_transform)                                                                                       

In [4]: a[0]                                                                                                                                               
called
Out[4]: 1

In [5]: a[0]                                                                                                                                               
called
Out[5]: 1
```

**Comment by guigautier** [comment]
Many thanks for your answer
